src/hotspot/share/nmt/vmatree.hpp

```diff
@@ -140,6 +142,14 @@ class VMATree {
 private:
   VMATreap _tree;
 
+  IntervalState& in_state(TreapNode* node) {
+    return node->val().in;
+  }
+
+  IntervalState& out_state(TreapNode* node) {
+    return node->val().out;
+  }
```

Comment:
They should be static and const, yes.

Re: ephemeral utility class. I do prefer using functions as opposed to methods, but it'd be nice to wrap them in a namespace or class that can be shortened and used outside of `VMATree`.

Reply:
Okay

Reply:
>Also, TreapNode* should be const.

Right, gotta back off from that again. The issue is that `val()` is non-const, which is correct: we should be able to change the value of a node. But, that means that this argument cannot be `const`, because (quoting my compiler`: `'this' argument discards qualifiers`. It's just an unfortunately clunky part of C++, it seems.